IBAnimatable/TimingFunctionType.swift

```diff
@@ -0,0 +1,355 @@
+//
+//  Created by phimage on 19/04/2017.
+//  Copyright © 2017 IBAnimatable. All rights reserved.
+//
+
+import UIKit
+
+public enum TimingFunctionType {
+  // well known tining functions
+  case linear
+  case easeIn
+  case easeOut
+  case easeInOut
+  case `default`
+
+  // not defined
+  case none
+
+  // custom one
+  case custom((x: Float, y: Float), (x: Float, y: Float))
+  case spring(damping: Float)
+
+  // from http://easings.net/
+  case easeInSine, easeOutSine, easeInOutSine
+  case easeInCubic, easeOutCubic, easeInOutCubic
+  case easeInQuad, easeOutQuad, easeInOutQuad
+  case easeInQuart, easeOutQuart, easeInOutQuart
+  case easeInQuint, easeOutQuint, easeInOutQuint
+  case easeInExpo, easeOutExpo, easeInOutExpo
+  case easeInCirc, easeOutCirc, easeInOutCirc
+  case easeInBack, easeOutBack, easeInOutBack
+
+}
+
+extension TimingFunctionType: IBEnum {
+
+  public init(string: String?) {
+    guard let string = string else {
+      self = .none
+      return
+    }
+
+    let (name, params) = TimingFunctionType.extractNameAndParams(from: string)
+
+    switch name {
+    // standards
+    case "linear":
+      self = .linear
+    case "easein":
+      self = .easeIn
+    case "easeout":
+      self = .easeOut
+    case "easeinout":
+      self = .easeInOut
+    case "default":
+      self = .default
+
+    // customs
+    case "spring" where params.count == 1:
+      self = .spring(damping: params[0].toFloat() ?? 0)
+    case "custom" where params.count == 4:
+      let c1 = (params[0].toFloat() ?? 0, params[1].toFloat() ?? 0)
+      let c2 = (params[2].toFloat() ?? 0, params[3].toFloat() ?? 0)
+      self = .custom(c1, c2)
+
+    case "easeinsine":
+      self = .easeInSine
+    case "easeoutsine":
+      self = .easeOutSine
+    case "easeinoutsine":
+      self = .easeInOutSine
+    case "easeinquad":
+      self = .easeInQuad
+    case "easeoutquad":
+      self = .easeOutQuad
+    case "easeinoutquad":
+      self = .easeInOutQuad
+    case "easeincubic":
+      self = .easeInCubic
+    case "easeoutcubic":
+      self = .easeOutCubic
+    case "easeinoutcubic":
+      self = .easeInOutCubic
+    case "easeinquart":
+      self = .easeInQuart
+    case "easeoutquart":
+      self = .easeOutQuart
+    case "easeinoutquart":
+      self = .easeInOutQuart
+    case "easeinquint":
+      self = .easeInQuint
+    case "easeoutquint":
+      self = .easeOutQuint
+    case "easeinoutquint":
+      self = .easeInOutQuint
+    case "easeinexpo":
+      self = .easeInExpo
+    case "easeoutexpo":
+      self = .easeOutExpo
+    case "easeinoutexpo":
+      self = .easeInOutExpo
+    case "easeincirc":
+      self = .easeInCirc
+    case "easeoutcirc":
+      self = .easeOutCirc
+    case "easeinoutcirc":
+      self = .easeInOutCirc
+    case "easeinback":
+      self = .easeInBack
+    case "easeoutback":
+      self = .easeOutBack
+    case "easeinoutback":
+      self = .easeInOutBack
+
+    default:
```

Comment:
Same here, I know it's already explicit, but I think it's better to keep the same structure as the enums. It helps to follow